packages/cdc-react/src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,58 @@
+import "./Modal.scss";
+import { Button } from "@us-gov-cdc/cdc-react";
+import { Icons } from "@us-gov-cdc/cdc-react-icons";
+import { useRef } from "react";
+export interface ModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  modalTitle: string;
+  children?: React.ReactNode;
+  isForcedAction?: boolean;
+}
+// export interface ModalChildrenProps {
+//   children: React.ReactNode;
+// }
+
+export const Modal = (props: ModalProps) => {
+  const overlayRef = useRef<HTMLDivElement>(null);
+  const modalRef = useRef<HTMLDivElement>(null);
+
+  const handleOverlayClick = (event: React.MouseEvent<HTMLDivElement>) => {
+    if (event.target === overlayRef.current && !props.isForcedAction) {
+      props.onClose();
+    }
+  };
+
+  if (!props.isOpen) return null;
+
+  return (
+    <div
+      ref={overlayRef}
+      onClick={handleOverlayClick}
+      role="presentation"
+      className="modal-overlay">
+      <div
+        ref={modalRef}
+        className="modal usa-modal"
+        role="dialog"
+        aria-modal="true">
+        <div className="modal-content usa-modal__content">
+          <div className="modal-main">
+            <div className="modal-header">
+              <h2 className="modal-title">{props.modalTitle}</h2>
+            </div>
+            {props.children}
+          </div>
+          <Button
```

Comment:
Let's move this button inside the modal header layout and control its position with flexbox instead of position absolute.

Reply:
Lets bring in Abbey to this discussion. She had mentioned how the close button needs to be the last element in the modal.

Reply:
Is this mentioned in the figma design?

Reply:
Abbey mentioned this during our refinement session on Oct 17. I asked about modal accessibility and we ended up making the modal story smaller in scope. She even added a comment in the jira ticket which links to the USWDS modal. She didn't provide context with her comment though. I reached out to her on teams and added you to the thread.